plugins/Login/tests/Integration/Security/LoginFromAnotherCountryTest.php

```diff
@@ -0,0 +1,172 @@
+    }
+
+    public function testNoEmailIsSentWhenLoggingInFirstTimeAfterRelease()
+    {
+        // start in France
+        $_SERVER['REMOTE_ADDR'] = self::IP_FRANCE;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+    }
+
+    public function testNoEmailIsSentWhenLoggingInFromTheSameCountryAsBefore()
+    {
+        // start in USA
+        $_SERVER['REMOTE_ADDR'] = self::IP_USA;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+
+        // continue in USA
+        $_SERVER['REMOTE_ADDR'] = self::IP_USA;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+    }
+
+    public function testEmailIsSentWhenLoggingInFromADifferentCountry()
+    {
+        // start in France
+        $_SERVER['REMOTE_ADDR'] = self::IP_FRANCE;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+
+        // continue in USA
+        $_SERVER['REMOTE_ADDR'] = self::IP_USA;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEquals(Piwik::translate('Intl_Country_US') ,$this->loginCountry);
+    }
+
+    public function testEmailIsSentWhenLoggingInFromAnUnknownCountry()
+    {
+        // start in France
+        $_SERVER['REMOTE_ADDR'] = self::IP_FRANCE;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+
+        // continue in an unknown country
+        $_SERVER['REMOTE_ADDR'] = self::IP_UNKNOWN;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEquals(Piwik::translate('General_Unknown') ,$this->loginCountry);
+
+        // continue in the USA again
+        $_SERVER['REMOTE_ADDR'] = self::IP_USA;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEquals(Piwik::translate('Intl_Country_US') ,$this->loginCountry);
+    }
+
+    public function emailIsNotSentWhenGeoIpNotAvailable()
+    {
+        LocationProvider::setCurrentProvider(LocationProvider\DisabledProvider::ID);
+
+        // start in France
+        $_SERVER['REMOTE_ADDR'] = self::IP_FRANCE;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+
+        // continue in USA
+        $_SERVER['REMOTE_ADDR'] = self::IP_USA;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+    }
+
+    public function emailIsNotSentWhenUsingDefaultGeoIpProvider()
```

Comment:
```suggestion
    public function testEmailIsNotSentWhenUsingDefaultGeoIpProvider()
```

Otherwise the test won't run.

Reply:
Good spot, fixed